Sources/ClientRuntime/Networking/Http/URLSession/FoundationStreamBridge.swift

```diff
@@ -202,7 +227,10 @@ class FoundationStreamBridge: NSObject, StreamDelegate {
             // Since space is available, try and read from the ReadableStream and
             // transfer the data to the Foundation stream pair.
             // Use a `Task` to perform the operation within Swift concurrency.
-            Task { try await writeToOutput() }
+            // If chunked transfer, writing to stream will be done manually
+            if !self.isChunkedTransfer {
+                Task { try await writeToOutput() }
+            }
```

Comment:
Im not familiar with how the stream bridge interacts with .hasSpaceAvailable so please correct me -- I added this because if I allow this writeToOutput() to be executed here then it is bypasses `sendAwsChunkedBody` and tries to read directly from the stream with `readAsync` followed by writing to the stream causing an error as that data is not in the form of a chunk with all the metadata (size, signature, checksum etc.)